### PR TITLE
Controllers plus middlewares

### PR DESCRIPTION
**Pull request by tamcio**

### CODE REVIEW

File: Server/middlewares/loginChecker.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Middleware for checking user authentication and decoding JWT tokens.
+ * 
+ * This middleware function that verifies the presence of a valid JWT token
+ * It decodes the token, checks its validity, and attaches
+ * the decoded user information to the request object if successful. If the token is missing or invalid,
+ * it returns an unauthorized request error.
+ */
+
+
+const loginChecker = asyncHandler(async (req, res, next) => {
```
Comment (JakKrzy) [inline Server/middlewares/loginChecker.js:16]: Would be more readable if if's were nested less
if ( !( req.headers.authorization...) {
   do else block
   return
if (! token)
   do else block
   return
jwt.verify(...)

File: Server/middlewares/roleChecker.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * 
+ * The roleChecker function returns middleware that checks whether a user has a specific role (admin or not).
+ * If the user has the required role, it passes the request to the next handler; otherwise, it returns an access denied error.
+ * 
+ * @param {boolean} bool - A boolean value indicating whether the user must be an administrator (true) or not (false).
+ * @returns {function} Middleware
+ * 
+ */
+
+const roleChecker = (bool) =>
```
Comment (JakKrzy) [inline Server/middlewares/roleChecker.js]: Better name for the param and it wouldn't have to be explained in comment ;)

**Review by JakKrzy — APPROVED** [review]
2 improvements left to do after the merge

File: Server/middlewares/loginChecker.js (hunk 1 above)
Comment (tamcio) [reply Server/middlewares/loginChecker.js:16]: Done

File: Server/middlewares/roleChecker.js (hunk 2 above)
Comment (tamcio) [reply Server/middlewares/roleChecker.js]: Done

**Pull request merged**